packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+          selectors.recommendationListItem
+        );
+        expect(recommendationListItems.length).toBe(0);
+      });
+    });
+
+    describe('when there are recommendations', () => {
+      describe('when the variant value is #grid', () => {
+        it('should properly display the recommendations with default options', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.gridContainer
+          );
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListItem
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationItems.length).toBe(recommendationsArray.length);
+        });
+
+        describe('with a custom #recommendationsPerRow option', () => {
+          const customRecommendationsPerRow = 2;
+          it('should set the width of the recommendation items based on the number of #recommendationsPerRow value', async () => {
+            const expectedRecommendationWidth = `${100 / customRecommendationsPerRow}%`;
+            const element = createTestComponent({
+              ...defaultOptions,
+              variant: 'grid',
+              recommendationsPerRow: 2,
+            });
+            await flushPromises();
+
+            expect(element.style._values['--recommendationItemWidth']).toEqual(
+              expectedRecommendationWidth
+            );
+          });
+        });
+
+        it('should properly pass the custom #label to the heading', async () => {
+          const customLabel = 'Custom label';
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+            label: customLabel,
+          });
+          await flushPromises();
+
+          const headingElement = element.shadowRoot.querySelector(
+            selectors.quanticHeading
+          );
+
+          expect(headingElement).not.toBeNull();
+          expect(headingElement.label).toEqual(customLabel);
+        });
+      });
+
+      describe('when the variant value is #carousel', () => {
+        it('should properly display the recommendations with default options', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'carousel',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.carouselContainer
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationContainer.label).toEqual(defaultOptions.label);
+          expect(recommendationContainer.itemsPerPage).toEqual(
+            defaultOptions.recommendationsPerRow
+          );
```

Comment:
same, why are we not looping over recommendationItems  to confirm that each recommendation has been correctly passed to a quantic result component?